scripts/autotools/libmongoc/CheckCompression.m4

```diff
@@ -47,7 +48,27 @@ AS_IF([test "$found_zlib" != "yes"],[
   bundled_zlib="yes"
 ])
 
-if test "$found_snappy" = "yes" -o "$found_zlib" = "yes" -o "$bundled_zlib" = "yes"; then
+PKG_CHECK_MODULES([PHP_MONGODB_ZSTD],[libzstd],[
+  PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS $PHP_MONGODB_ZSTD_CFLAGS"
+  PHP_EVAL_LIBLINE([$PHP_MONGODB_ZSTD_LIBS],[MONGODB_SHARED_LIBADD])
+  found_zstd="yes"
+],[
+  PHP_CHECK_LIBRARY([zstd],
+                    [ZSTD_compress],
```

Comment:
It was trivial to copy the fallback detection when `pkg-config` is not available. This API comes from the [zstd documentation](http://facebook.github.io/zstd/zstd_manual.html) and you'll find the example closely resembles what we do for snappy above.

One difference is that the `pkg-config` name is "libzstd", while the shared library itself is simply "zstd".